### PR TITLE
"new Buffer" Deprecated changed to Buffer.from

### PR DESCRIPTION
**Pull request by avbanks**
Hey, I just changed the usage of new Buffer to Buffer.from If I'm not mistaken from the docs it's a drop-in replacement.  Node gives a stability index of 0 for new Buffer. The change seems fine from my testing, can anyone else take a look?

https://nodejs.org/api/buffer.html#buffer_class_method_buffer_from_buffer

### CODE REVIEW

**Comment by alexbarnsley** [comment]
Hi @avbanks I had an issue when I connected Ledger:

```
$ node index.js 
    ___         __      _________            __ 
   /   |  _____/ /__   / ____/ (_)__  ____  / /_
  / /| | / ___/ //_/  / /   / / / _ \/ __ \/ __/
 / ___ |/ /  / ,<    / /___/ / /  __/ / / / /_  
/_/  |_/_/  /_/|_|   \____/_/_/\___/_/ /_/\__/  
                                                
ark> connect devnet
{ nethash: '578e820911f24e039733b45e4882b73e301f813a0d2c31330dafda84534ffa23',
  token: 'DARK',
  symbol: 'DѦ',
  explorer: 'http://dexplorer.ark.io',
  version: 30 }
Node: 167.114.29.55:4002, height: 2457288
ark devnet> no request: TypeError [ERR_INVALID_ARG_TYPE]: The "value" argument must not be of type number. Received type number
    at Function.from (buffer.js:208:11)
    at LedgerArk.getAddress_async (/Users/alexbarnsley/projects/ark-client/src/LedgerArk.js:13:22)
    at populateLedgerAccounts (/Users/alexbarnsley/projects/ark-client/index.js:264:26)
    at ledger.comm_node.create_async.then (/Users/alexbarnsley/projects/ark-client/index.js:346:7)
    at _fulfilled (/Users/alexbarnsley/projects/ark-client/node_modules/ledgerco/node_modules/q/q.js:834:54)
    at self.promiseDispatch.done (/Users/alexbarnsley/projects/ark-client/node_modules/ledgerco/node_modules/q/q.js:863:30)
    at Promise.promise.promiseDispatch (/Users/alexbarnsley/projects/ark-client/node_modules/ledgerco/node_modules/q/q.js:796:13)
    at /Users/alexbarnsley/projects/ark-client/node_modules/ledgerco/node_modules/q/q.js:604:44
    at runSingle (/Users/alexbarnsley/projects/ark-client/node_modules/ledgerco/node_modules/q/q.js:137:13)
    at flush (/Users/alexbarnsley/projects/ark-client/node_modules/ledgerco/node_modules/q/q.js:125:13)
```

Are you able to reproduce?

**Comment by avbanks** [comment]
Hey, @alexbarnsley what version of Node are you using? I'm connecting to devnet with no issues on v9.4.0

**Comment by alexbarnsley** [comment]
@avbanks I'd say the same 🤔  i'll give it another go later, make sure it wasn't my setup

```
Alexs-MBP: alexbarnsley$ node -v
v9.3.0
Alexs-MBP: alexbarnsley$ npm -v
5.6.0
```

**Comment by faustbrian** [comment]
@alexbarnsley any updates on this?

**Comment by alexbarnsley** [comment]
@avbanks okay, i'm still getting an error for some reason, but at a different point in time now. Here's what happened:

```
Alexs-MBP:ark-core alexbarnsley$ ark-client
    ___         __      _________            __ 
   /   |  _____/ /__   / ____/ (_)__  ____  / /_
  / /| | / ___/ //_/  / /   / / / _ \/ __ \/ __/
 / ___ |/ /  / ,<    / /___/ / /  __/ / / / /_  
/_/  |_/_/  /_/|_|   \____/_/_/\___/_/ /_/\__/  
                                                
ark> connect devnet
{ nethash: '578e820911f24e039733b45e4882b73e301f813a0d2c31330dafda84534ffa23',
  token: 'DARK',
  symbol: 'DѦ',
  explorer: 'http://dexplorer.ark.io',
  version: 30 }
Node: 167.114.29.53:4002, height: 2576100
Ledger App Connected
ark devnet> account send 1 D6GuEAxkJgAwAmTNPc6bf2RFYe4q9f9FSD
We have found the following Ledgers: 
1) D7u5Ro3WihcbWFgxSGcqpqJypVeaLpfFw7 (DѦ509.59999997)
N) passphrase

Please choose an option: 1
? Sending 1DARK to D6GuEAxkJgAwAmTNPc6bf2RFYe4q9f9FSD now Yes
(node:37638) UnhandledPromiseRejectionWarning: TypeError: Illegal value: undefined (not an integer)
    at ByteBuffer.module.exports.ByteBufferPrototype.writeInt8 (/usr/local/lib/node_modules/ark-client/node_modules/bytebuffer/dist/bytebuffer-node.js:533:23)
    at Object.getBytes (/usr/local/lib/node_modules/ark-client/node_modules/arkjs/lib/transactions/crypto.js:99:5)
    at ledgerSignTransaction (/usr/local/lib/node_modules/ark-client/index.js:322:37)
    at /usr/local/lib/node_modules/ark-client/index.js:866:13
    at cb (/usr/local/lib/node_modules/ark-client/node_modules/vorpal/dist/vorpal.js:552:9)
    at /usr/local/lib/node_modules/ark-client/node_modules/vorpal/dist/vorpal.js:573:9
    at module.exports.completed (/usr/local/lib/node_modules/ark-client/node_modules/vorpal/dist/ui.js:220:13)
    at module.exports.PromptUI.onCompletion (/usr/local/lib/node_modules/ark-client/node_modules/inquirer/lib/ui/prompt.js:57:10)
    at AnonymousObserver.Rx.AnonymousObserver.AnonymousObserver.completed (/usr/local/lib/node_modules/ark-client/node_modules/rx-lite/rx.lite.js:1550:12)
    at AnonymousObserver.Rx.internals.AbstractObserver.AbstractObserver.onCompleted (/usr/local/lib/node_modules/ark-client/node_modules/rx-lite/rx.lite.js:1489:14)
(node:37638) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:37638) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**Edit:** This is consistent and i get it every time i do the above process.

**Comment by avbanks** [comment]
@Alexbarnsley I’ll take a look and resolve the issue this weekend.

**Comment by alexbarnsley** [comment]
@avbanks no worries. Maybe someone else could give it a try. @faustbrian, have you got a ledger?

File: src/LedgerArk.js [hunk 1]
```diff
@@ -91,15 +91,15 @@ LedgerArk.prototype.signTransaction_async = function(path, rawTxHex) {
 }
 
 LedgerArk.prototype.getAppConfiguration_async = function() {
-	var buffer = new Buffer(5);
+	var buffer = Buffer.from(5);
```
Comment (faustbrian) [inline src/LedgerArk.js:94]: If you wish to create a new Buffer of a specific length you need to use `Buffer.alloc`

File: src/LedgerArk.js [hunk 2]
```diff
@@ -109,14 +109,14 @@ LedgerArk.prototype.getAppConfiguration_async = function() {
...
-	var message = new Buffer(messageHex, 'hex');
+	var message = Buffer.from(messageHex, 'hex');
 	var apdus = [];
 	var response = [];
 	var self = this;
 	while (offset != message.length) {
 		var maxChunkSize = (offset == 0 ? (150 - 1 - splitPath.length * 4 - 4) : 150)
 		var chunkSize = (offset + maxChunkSize > message.length ? message.length - offset : maxChunkSize);
-		var buffer = new Buffer(offset == 0 ? 5 + 1 + splitPath.length * 4 + 4 + chunkSize : 5 + chunkSize);
+		var buffer = Buffer.from(offset == 0 ? 5 + 1 + splitPath.length * 4 + 4 + chunkSize : 5 + chunkSize);
```
Comment (faustbrian) [inline src/LedgerArk.js:119]: If you wish to create a new Buffer of a specific length you need to use `Buffer.alloc`.

File: src/LedgerArk.js [hunk 3]
```diff
@@ -34,18 +34,18 @@ LedgerArk.prototype.getAddress_async = function(path) {
...
 	var splitPath = utils.splitPath(path);
-	var rawTx = new Buffer(rawTxHex, 'hex');
+	var rawTx = Buffer.from(rawTxHex, 'hex');
 	var self = this;
 
 	var data1, data2;
-	var data1_headerlength = new Buffer(2);
-	var data2_headerlength = new Buffer(1);
+	var data1_headerlength = Buffer.from(2);
+	var data2_headerlength = Buffer.from(1);
```
Comment (faustbrian) [inline src/LedgerArk.js:42]: If you wish to create a new Buffer of a specific length you need to use `Buffer.alloc`.

File: src/LedgerArk.js [hunk 4]
```diff
@@ -34,18 +34,18 @@ LedgerArk.prototype.getAddress_async = function(path) {
...
 LedgerArk.prototype.signTransaction_async = function(path, rawTxHex) {
 	var splitPath = utils.splitPath(path);
-	var rawTx = new Buffer(rawTxHex, 'hex');
+	var rawTx = Buffer.from(rawTxHex, 'hex');
 	var self = this;
 
 	var data1, data2;
-	var data1_headerlength = new Buffer(2);
-	var data2_headerlength = new Buffer(1);
+	var data1_headerlength = Buffer.from(2);
```
Comment (faustbrian) [inline src/LedgerArk.js:41]: If you wish to create a new Buffer of a specific length you need to use `Buffer.alloc`.

File: src/LedgerArk.js [hunk 5]
```diff
@@ -10,7 +10,7 @@ var LedgerArk = function(comm) {
 
 LedgerArk.prototype.getAddress_async = function(path) {
 	var splitPath = utils.splitPath(path);
-	var buffer = new Buffer(5 + 1 + splitPath.length * 4);
+	var buffer = Buffer.from(5 + 1 + splitPath.length * 4);
```
Comment (faustbrian) [inline src/LedgerArk.js:13]: If you wish to create a new Buffer of a specific length you need to use `Buffer.alloc`.

**Comment by faustbrian** [comment]
@alexbarnsley No but I added some comments to the code that probably are the issue.

**Comment by alexbarnsley** [comment]
@faustbrian thanks! Those are the lines I saw based on the errors too, but don't know enough about Buffer to know what's right without making changes myself 👌

**Comment by faustbrian** [comment]
You should be able to just replace `Buffer.from` with `Buffer.alloc` in all mentioned lines and give it another try.

**Comment by faustbrian** [comment]
Closing this because of lack of response.

**Pull request closed without merging**